overlord/snapstate/handlers.go

```diff
@@ -1235,11 +1292,13 @@ func writeSeqFile(name string, snapst *SnapState) error {
 	}
```

Comment:
we use writeSeqFile it seems after linking the snap (backend.LinkSnap etc),  it seems we might need to move the write or have a write before and after. Otherwise we might have app from the snap running before the right value for MigrateHidden has been exposed?

Reply:
That's true, I think we can write just once before linking as long we revert it if `doLinkSnap` fails afterwards. On master we currently don't do that, but I don't see an explicit reason for that. AFACT, we should also revert the sequence and current if the install/update fails

Reply:
Just to correct myself: we do revert the sequence file, we just do it in the undoLinkSnap https://github.com/snapcore/snapd/blob/47b19f2a27ee4eaee24d3f0b370bc1d34c8582bc/overlord/snapstate/handlers.go#L1927 
Having to revert the seq file is also racy because the snap might run in between seq file writes. I'll go with writing twice to avoid this